extensions/wikia/TemplateClassification/TemplateClassification.hooks.php

```diff
@@ -15,7 +14,96 @@ public static function register() {
 		\Hooks::register( 'BeforePageDisplay', [ $hooks, 'onBeforePageDisplay' ] );
 		\Hooks::register( 'PageHeaderPageTypePrepared', [ $hooks, 'onPageHeaderPageTypePrepared' ] );
 		\Hooks::register( 'QueryPageUseResultsBeforeRecache', [ $hooks, 'onQueryPageUseResultsBeforeRecache' ] );
+		/* Edit page hooks */
+		\Hooks::register( 'EditPage::showEditForm:fields', [ $hooks, 'onEditPageShowEditFormFields' ] );
+		\Hooks::register( 'ArticleInsertComplete', [ $hooks, 'onArticleInsertComplete' ] );
 		\Hooks::register( 'EditPageLayoutExecute', [ $hooks, 'onEditPageLayoutExecute' ] );
+		\Hooks::register( 'EditPageMakeGlobalVariablesScript', [ $hooks, 'onEditPageMakeGlobalVariablesScript' ] );
+	}
+
+	/**
+	 * Save template type passed from article creation
+	 * template type is stored in templateClassificationType hidden field
+	 *
+	 * @param \WikiPage $wikiPage
+	 * @return bool
+	 */
+	public function onArticleInsertComplete( \WikiPage $wikiPage ) {
+		global $wgCityId;
+
+		$request = \RequestContext::getMain()->getRequest();
+		$type = $request->getVal( 'templateClassificationType' );
+
+		/**
+		 * The service was not available when the field's value was set
+		 * so we exit early to prevent polluting of the results.
+		 */
+		if ( $type === \TemplateClassificationService::NOT_AVAILABLE ) {
+			return true;
+		}
+
+		try {
+			( new \TemplateClassificationService() )->classifyTemplate(
+				$wgCityId,
+				$wikiPage->getId(),
+				$type,
+				\TemplateClassificationService::USER_PROVIDER,
+				$wikiPage->getUser()
+			);
+		} catch ( \Exception $e ) {
```

Comment:
We had this discussion yesterday and the `insertTemplateDetails` throws either an `InvalidArgumentException` (pre-request) or a `Swagger\Client\ApiException` (post-request). Both classes have only one common parent - the generic `Exception` class. If we do not catch the `InvalidArgumentException` it will in fact fail loudly. Given how the methods are used (in hooks and when rendering a page) I'd actually consider the `InvalidArgumentException` worth it. Thank you!